### PR TITLE
Add back de novo caller!

### PR DESCRIPTION
**Pull request by tpoterba**
Resolves https://github.com/hail-is/hail/pull/2471 by reimplementing in Python.

Required changes to bind to permit binding multiple args; this required flipping the order of the function and the binding.

You shouldn't need to do much correctness checking in the implementation, because the files used to test basically catch every edge case.

### CODE REVIEW

**Comment by tpoterba** [comment]
I had to change the original VCF to put the X / Y coordinates in the GRCh37 range, though (I just interpolated between the correct start + end)

File: python/hail/methods/family_methods.py [hunk 1]
```diff
@@ -386,3 +387,365 @@ def transmission_disequilibrium_test(dataset, pedigree) -> Table:
...
+
+    This method produces a :class:`.Table` with the following fields:
+
+     - `locus` (``locus``) -- Variant locus.
+     - `alleles` (``array<str>``) -- Variant alleles.
+     - `prior` (``float64``) -- Site frequency prior. It is the maximum of:
+       the computed dataset alternate allele frequency, the
+       `pop_frequency_prior` parameter, and the global prior
+       ``1 / 3e7``.
+     - `proband` (``struct``) -- Proband column fields.
```
Comment (catoverdrive) [inline python/hail/methods/family_methods.py]: all of the ones from the original dataset?

File: python/hail/methods/family_methods.py [hunk 2]
```diff
@@ -386,3 +387,365 @@ def transmission_disequilibrium_test(dataset, pedigree) -> Table:
...
+     - `father` (``struct``) -- Father column fields.
+     - `mother` (``struct``) -- Mother column fields.
+     - `proband_entry` (``struct``) -- Proband entry fields.
+     - `father_entry` (``struct``) -- Father entry fields.
+     - `proband_entry` (``struct``) -- Mother entry fields.
+     - `is_female` (``bool``) -- Proband is female.
```
Comment (catoverdrive) [inline python/hail/methods/family_methods.py]: "Whether or not..." or "If `True`, ...", probably?

File: python/hail/methods/family_methods.py (hunk 1 above)
Comment (catoverdrive) [reply python/hail/methods/family_methods.py]: (it feels a little bit unclear where they're coming from)

File: python/hail/expr/functions.py [hunk 3]
```diff
@@ -276,8 +277,8 @@ def switch(expr) -> 'hail.expr.builders.SwitchBuilder':
     return SwitchBuilder(expr)
 
 
-@typecheck(expr=expr_any, f=func_spec(1, expr_any))
-def bind(expr, f: Callable):
+@typecheck(f=anytype, exprs=expr_any)
+def bind(f: Callable, *exprs):
```
Comment (catoverdrive) [inline python/hail/expr/functions.py]: can you add some tests for multiple exprs?
